src/Nethermind/Nethermind.Db/Blooms/FixedSizeFileStore.cs

```diff
@@ -58,41 +53,14 @@ public void Write(long index, ReadOnlySpan<byte> element)
 
         public int Read(long index, Span<byte> element)
         {
-            EnsureFlushed();
-
-            lock (_fileRead)
-            {
-                SeekIndex(_fileRead, index);
-                return _fileRead.Read(element);
-            }
+            return RandomAccess.Read(_fileRead, element, GetPosition(index));
```

Comment:
a bit worried here, previous lock+flush guaranteed here that all previous Write's would already be visible for this Read, would this still be the case?

Reply:
Should still be the case; previous flush was emptying the internal .NET `FileStream` buffers but not forcing the OS disk flush. Random access is a straight OS write so isn't storing anything internally so doesn't need that flush.

Since everything is offset driven not sure the intermediate `FileStream` buffer is gaining anything as its moving about all the time? i.e. Random Access

Reply:
Changed to use same handle for reading and writing; and also parallalize the access since it operates over 4 files at a time